src/type/__tests__/validation-test.js

```diff
@@ -896,6 +896,194 @@ describe('Type System: Objects can only implement unique interfaces', () => {
   });
 });
 
+describe('Type System: Interface extensions should be valid', () => {
+  it('rejects an Object implementing the extended interface due to missing field', () => {
+    const schema = buildSchema(`
+      type Query {
+        test: AnotherObject
+      }
+
+      interface AnotherInterface {
+        field: String
+      }
+
+      type AnotherObject implements AnotherInterface {
+        field: String
+      }
+    `);
+    const extendedSchema = extendSchema(
+      schema,
+      parse(`
+        extend interface AnotherInterface {
+          newField: String
+        }
+      `),
+    );
+    expect(validateSchema(extendedSchema)).to.containSubset([
+      {
+        message:
+          'Interface field AnotherInterface.newField expected but AnotherObject does not provide it.',
+        locations: [{ line: 3, column: 11 }, { line: 10, column: 7 }],
+      },
+    ]);
+  });
+
+  it('accepts an Object implementing the extended interface with existing field', () => {
```

Comment:
I think it's more appropriate to have this test as part of `extendSchema` test suite.